src/shared/cloudformation/templateRegistry.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import { fileURLToPath, pathToFileURL } from 'url'
+import * as vscode from 'vscode'
+import { FileWatcherListener } from '../utilities/fileSystemWatcher'
+import { CloudFormation } from './cloudformation'
+
+let templateRegistry: CloudFormationTemplateRegistry | undefined
+
+export function getTemplateRegistry(): CloudFormationTemplateRegistry {
+    if (!templateRegistry) {
+        throw new Error(
+            'Template Registry not initialized. Extension code should call activate() from shared/cloudformation/activation, test code should call setTemplateRegistry().'
+        )
+    }
+
+    return templateRegistry
+}
+
+export function setTemplateRegistry(registry: CloudFormationTemplateRegistry | undefined): void {
+    templateRegistry = registry
+}
+
+export interface CloudFormationTemplateRegistry {
```

Comment:
why is this an interface? I don't understand why everything needs to be polymorphic. It's a two-way door: add classes/interfaces when you need them or if you know you will need them. But otherwise just don't add them, it's noise.

Reply:
For the most part, this has been our convention for testing purposes (e.g. this was written to match the conventions put in place by the `Logger`). @awschristou do you have any thoughts on this?

Reply:
I'm inclined to think that an interface is not necessary. We don't intend to have an alternate implementation for this, and testing can leverage a mock where needed.

Reply:
> for testing purposes

Does that mean "mocks" or something else? If it's for mocks, then let's delete the mock framework _and_ the interfaces :)